runtime/gc_glue_java/ObjectModel.cpp

```diff
@@ -120,7 +136,25 @@ GC_ObjectModel::internalClassLoadHook(J9HookInterface** hook, UDATA eventNum, vo
 			objectModel->_classClass = clazz;
 		} else if (0 == compareUTF8Length(J9UTF8_DATA(className), J9UTF8_LENGTH(className), (U_8*)abstractOwnableSynchronizer, strlen(abstractOwnableSynchronizer))) {
 			 clazz->classDepthAndFlags |= J9AccClassOwnableSynchronizer;
-		}
+		} else if (0 == compareUTF8Length(J9UTF8_DATA(className), J9UTF8_LENGTH(className), (U_8*)javaLangString, strlen(javaLangString))) {
+			 clazz->hotFieldOffset =  1; 
```

Comment:
How will this code handle changes to the layout of the classes?  What if we or OpenJDK refactored the class and reordered the fields?  Less of an issue for string but more treemap entry, hashmaps, should expect that the layouts may be refactored.

Reply:
You are correct this code will not be able to handle changes to the layout of the classes. To resolve this it would be need to be done either dynamically through identifying the field name and setting it based off the field name (would fail if the field names were refactored) or perhaps this can be scrapped and the algorithm in VMJ9.cpp should be trusted to set the correct hot field? (this would work for all the classes except for string as strings do not see my code in VMJ9.cpp, the only penalty would be in performance which I do not believe would be substantial, but still should be measure)

Reply:
Would you have thoughts regarding this @amicic?